lib/cloudant.js

```diff
@@ -524,20 +531,40 @@ Cloudant.prototype.updateAll = function(model, where, data, options, cb) {
           model, where, data, options);
   var self = this;
   var mo = self.selectModel(model);
+  var resultArray = [];
   self.all(model, {where: where}, {raw: true}, function(err, docs) {
     if (err) return cb(err, docs);
-    if (docs.length === 0) return cb(null, {count: 0});
+    var conflicts = [];
+    if (docs.length === 0) {
+      err = new Error(g.f('No document found with filter: %j%s%s',
+        where, os.EOL, conflicts.join(os.EOL)));
+      err.statusCode = 404;
+      return cb(err);
+    }
+    _.each(docs, function(doc) {
+      if (data._rev) {
+        if (data._rev !== doc._rev) conflicts.push(doc._id);
```

Comment:
There is a bulk update method provided by the cloudant driver, I think we should use that method to `updateAll`.
> `if (data._rev !== doc._rev) conflicts.push(doc._id);`

would not be a valid check for an array of instances, each instance has its own `_rev`, we could not expect user provide a single `_rev` that could update all instances....

Haven't figured out what is the proper way to do a bulk update with `_rev` provided in data, I will double check juggler code :(